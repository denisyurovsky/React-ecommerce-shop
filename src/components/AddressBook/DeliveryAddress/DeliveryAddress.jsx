import { Box, TextField } from '@mui/material';
import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { toast } from 'react-toastify';

import { notificationError } from '../../../constants/constants';
import validateField from '../../../helpers/validateField';
import usePrevious from '../../../hooks/usePrevious';
import { RUSSIA } from '../../../pages/AddressBookPage/constants/constants';
import addressType from '../../../propTypes/addressType';
import { getCities, selectCities } from '../../../store/cities/citiesSlice';
import {
  getCountries,
  selectCountries,
} from '../../../store/countries/countriesSlice';
import CountryAutocomplete from '../CountryAutocomplete/CountryAutocomplete';
import formatRussianZip from '../helpers/formatRussianZip';
import VirtualizedAutocomplete from '../VirtualizedAutocomplete/VirtualizedAutocomplete.jsx';

import styles from './DeliveryAddress.module.scss';

const DeliveryAddress = ({ handleChange, address, setAddress }) => {
  const {
    country,
    zip,
    street,
    building,
    flat,
    country: { name: countryName, id: countryCode },
  } = address;
  const countries = useSelector(selectCountries);
  const cities = useSelector(selectCities);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCountries());
  }, [dispatch]);

  useEffect(() => {
    dispatch(getCities(countryCode));
  }, [countryCode, dispatch]);

  const { errorOccurred: citiesErrorOccurred } = cities;
  const { errorOccurred: countriesErrorOccurred } = countries;
  const prevAmount = usePrevious({
    citiesErrorOccurred,
    countriesErrorOccurred,
  });

  useEffect(() => {
    const isErrorOccurred = countriesErrorOccurred || citiesErrorOccurred;

    if (prevAmount) {
      const isErrorOccurredPrev =
        prevAmount.citiesErrorOccurred || prevAmount.countriesErrorOccurred;

      if (isErrorOccurredPrev !== isErrorOccurred && isErrorOccurred) {
        toast.error(notificationError);
      }
    }
  }, [countriesErrorOccurred, citiesErrorOccurred, prevAmount]);

  const [isStreetValid, setIsStreetValid] = useState(true);
  const [isBuildingValid, setIsBuildingValid] = useState(true);
  const [isZipValid, setIsZipValid] = useState(true);

  const handleChangeCountry = (e, newValue) => {
    const country = newValue === null ? { name: '', id: '' } : newValue;
    const formattedZip = country.name === RUSSIA ? formatRussianZip(zip) : zip;

    setAddress({
      ...address,
      country,
      city: '',
      zip: formattedZip,
    });
  };

  const handleChangeZip = (e) => {
    const { value } = e.target;
    const zip = countryName === RUSSIA ? formatRussianZip(value) : value;

    setAddress({ ...address, zip });
  };

  const validateZipField = () => {
    if (countryName === RUSSIA) {
      setIsZipValid(zip.length === 6);
    } else {
      setIsZipValid(Boolean(zip));
    }
  };

  const validateStreetField = (e) => {
    validateField(e, setIsStreetValid);
  };
  const suspendStreetFieldValidation = () => {
    setIsStreetValid(true);
  };
  const validateBuildingField = (e) => {
    validateField(e, setIsBuildingValid);
  };
  const suspendBuildingFieldValidation = () => {
    setIsBuildingValid(true);
  };
  const suspendZipFieldValidation = () => {
    setIsZipValid(true);
  };

  return (
    <>
      <Box className={styles.wrapper}>
        <CountryAutocomplete
          className={styles.country}
          handleChangeCountry={handleChangeCountry}
          country={country}
          countries={countries}
        />
        <VirtualizedAutocomplete
          setAddress={setAddress}
          address={address}
          data={cities.data}
          label="City"
        />
      </Box>
      <TextField
        fullWidth
        onChange={handleChange}
        id="street"
        value={street}
        label="Street"
        variant="standard"
        onBlur={validateStreetField}
        onFocus={suspendStreetFieldValidation}
        error={!isStreetValid}
        helperText={isStreetValid ? ' ' : 'This field should not be empty'}
      />
      <Box className={styles.wrapper}>
        <TextField
          className={styles.building}
          onChange={handleChange}
          id="building"
          value={building}
          label="Building"
          variant="standard"
          onBlur={validateBuildingField}
          onFocus={suspendBuildingFieldValidation}
          error={!isBuildingValid}
          helperText={isBuildingValid ? ' ' : 'This field should not be empty'}
        />
        <TextField
          onChange={handleChange}
          id="flat"
          value={flat}
          label="Flat"
          variant="standard"
        />
      </Box>
      <TextField
        fullWidth
        label="Zip"
        variant="standard"
        value={zip}
        onChange={handleChangeZip}
        onBlur={validateZipField}
        onFocus={suspendZipFieldValidation}
        error={!isZipValid}
        helperText={isZipValid ? ' ' : 'The zip is invalid'}
      />
    </>
  );
};

DeliveryAddress.propTypes = {
  handleChange: PropTypes.func.isRequired,
  setAddress: PropTypes.func.isRequired,
  address: addressType.isRequired,
};

export default DeliveryAddress;
