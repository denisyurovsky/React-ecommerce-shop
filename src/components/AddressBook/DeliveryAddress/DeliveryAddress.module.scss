@import "../../../assets/styles/variables.scss";

.wrapper {
  display: flex;
  flex-grow: 1;

  & > * {
    width: 50%;
  }
}

.country,
.building {
  margin-right: 10px !important;
}
