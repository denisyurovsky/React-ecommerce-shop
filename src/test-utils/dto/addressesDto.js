export const newAddress = {
  title: 'Mr.',
  name: 'Jon',
  surname: 'Smith',
  country: { id: 'RU', name: 'Russian Federation' },
  city: 'Saint Petersburg',
  street: 'Nevskiy',
  building: '8',
  flat: '33',
  phone: '+7 (981) 1112233',
  zip: '168142',
};

export const addresses = [
  {
    id: 1,
    title: 'Mr.',
    name: 'Alexander',
    surname: 'Sidorov',
    country: {
      id: 'RU',
      name: 'Russian Federation',
    },
    city: 'Saint Petersburg',
    street: 'Pulkovskaya',
    building: '8',
    flat: '4',
    phone: '+7 (981) 3002020',
    zip: '168142',
  },
  {
    id: 2,
    title: 'Mr.',
    name: 'Ivan',
    surname: 'Ivanov',
    country: {
      id: 'RU',
      name: 'Russian Federation',
    },
    city: 'Saint Petersburg',
    street: 'Leninskiy',
    building: '8',
    flat: '4',
    phone: '+7 (981) 1112233',
    zip: '168142',
  },
  {
    id: 3,
    title: 'Mrs.',
    name: 'Olga',
    surname: 'Petrova',
    country: {
      id: 'RU',
      name: 'Russian Federation',
    },
    city: 'Saint Petersburg',
    street: 'Pobedy',
    building: '8',
    flat: '4',
    phone: '+7 (981) 1233232',
    zip: '168142',
  },
  {
    id: 4,
    ...newAddress,
  },
  {
    id: 5,
    title: 'Mrs.',
    name: 'Elon',
    surname: 'Musk',
    country: { id: 'RU', name: 'Russian Federation' },
    city: 'Saint Petersburg',
    street: 'Nevskiy',
    building: '8',
    flat: '33',
    phone: '+7 (981) 1112233',
    zip: '168142',
    createdAt: '2021-04-10T10:16:28.404Z',
    updatedAt: '2021-12-28T23:24:21.552Z',
  },
];

export const addressesDto = [
  {
    id: 0,
    title: 'Mr.',
    name: 'Alexander',
    surname: 'Sidorov',
    country: {
      id: 'RU',
      name: 'Russian Federation',
    },
    city: 'Saint Petersburg',
    street: 'Pulkovskaya',
    building: '8',
    flat: '4',
    phone: '+7 (981) 3002020',
    zip: '168142',
    createdAt: '2021-04-10T10:16:28.404Z',
    updatedAt: '2021-12-28T23:24:21.552Z',
  },
  {
    id: 2,
    title: 'Mrs.',
    name: 'Alexandra',
    surname: 'Sidorova',
    country: {
      id: 'RU',
      name: 'Russian Federation',
    },
    city: 'Saint Petersburg',
    street: 'Pulkovskaya',
    building: '8',
    flat: '4',
    phone: '+7 (981) 1233232',
    zip: '168142',
    createdAt: '2021-09-07T15:27:43.842Z',
    updatedAt: '2021-10-21T22:59:08.494Z',
  },
];
