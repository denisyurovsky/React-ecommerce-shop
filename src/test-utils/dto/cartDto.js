export const testCart = {
  sellers: {
    1: {
      products: [
        {
          userId: 1,
          productId: 36,
          quantity: 1,
          checked: true,
          price: 724,
          discountPrice: null,
        },
      ],
      checked: true,
    },
    3: {
      products: [
        {
          userId: 3,
          productId: 23,
          quantity: 1,
          checked: true,
          price: 891,
          discountPrice: null,
        },
        {
          userId: 3,
          productId: 28,
          quantity: 1,
          checked: true,
          price: 210,
          discountPrice: 100,
        },
        {
          userId: 3,
          productId: 6,
          quantity: 1,
          checked: true,
          price: 510,
          discountPrice: 400,
        },
      ],
      checked: true,
    },
    4: {
      products: [
        {
          userId: 4,
          productId: 10,
          quantity: 1,
          checked: true,
          price: 757,
          discountPrice: null,
        },
      ],
      checked: true,
    },
  },
  totalPrice: 3092,
  totalDiscountPrice: 2872,
  totalQuantity: 5,
  isLoading: false,
  errorOccurred: false,
};

export const checkoutCart = {
  sellers: {
    1: {
      products: [
        {
          userId: 1,
          productId: 10,
          quantity: 1,
          checked: true,
          price: 917,
          discountPrice: 679,
        },
        {
          userId: 1,
          productId: 24,
          quantity: 1,
          checked: true,
          price: 108,
          discountPrice: 27,
        },
      ],
      checked: true,
    },
    2: {
      products: [
        {
          userId: 2,
          productId: 50,
          quantity: 1,
          checked: true,
          price: 787,
          discountPrice: null,
        },
        {
          userId: 2,
          productId: 33,
          quantity: 2,
          checked: true,
          price: 40,
          discountPrice: null,
        },
      ],
      checked: true,
    },
  },
  totalPrice: 1892,
  totalDiscountPrice: 1573,
  totalQuantity: 5,
  isLoading: false,
  errorOccurred: false,
  errorMessage: '',
};
