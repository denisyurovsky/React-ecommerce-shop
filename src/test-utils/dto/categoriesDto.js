const categoriesDto = [
  'All categories',
  'Grocery',
  'Industrial',
  'Clothing',
  'Computers',
];

export default categoriesDto;
