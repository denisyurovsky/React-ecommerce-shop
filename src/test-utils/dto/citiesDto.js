export const citiesRU = {
  cities: [
    'Krasnolesnyy',
    'Moscow',
    'Indura',
    'Sosnovka',
    'Saint Petersburg',
    'Zykovo',
    'Zyablikovo',
    'Zyukayka',
  ],
};

export const citiesBY = {
  cities: [
    'Kosterevo',
    'Minsk',
    'Sosnovka',
    'Skidel’',
    'Horad Zhodzina',
    'Zhlobin',
    'Zyuzino',
    'Zel’va',
    'Zaslawye',
  ],
};
