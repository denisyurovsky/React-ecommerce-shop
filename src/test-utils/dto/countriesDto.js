export const countries = [
  { id: 'AU', name: 'Australia' },
  { id: 'AT', name: 'Austria' },
  { id: 'BY', name: 'Belarus' },
  { id: 'BE', name: 'Belgium' },
  { id: 'BM', name: 'Bermuda' },
  { id: 'RU', name: 'Russian Federation' },
];

export const objCountries = {
  data: countries,
  isLoading: false,
  errorOccurred: false,
  errorMessage: '',
};
