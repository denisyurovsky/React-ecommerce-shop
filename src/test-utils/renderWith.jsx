import { configureStore } from '@reduxjs/toolkit';
import { render as rtlRender } from '@testing-library/react';
import PropTypes from 'prop-types';
import React from 'react';
import { Provider } from 'react-redux';
import { ToastContainer } from 'react-toastify';

import cartReducer from '../store/cart/cartSlice';
import categoriesReducer from '../store/categories/categoriesSlice';
import feedbackReducer from '../store/feedback/feedbackSlice';
import productsReducer from '../store/products/productsSlice';
import userReducer from '../store/user/userSlice';

import RouterConnected from './RouterConnected';

function render(
  component,
  preloadedState = {},
  path = '/',
  initialPaths = ['/']
) {
  const store = configureStore({
    reducer: {
      cart: cartReducer,
      products: productsReducer,
      categories: categoriesReducer,
      user: userReducer,
      feedback: feedbackReducer,
    },
    preloadedState,
  });

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <RouterConnected
          component={children}
          path={path}
          initialPaths={initialPaths}
        />
        <ToastContainer />
      </Provider>
    );
  }

  Wrapper.propTypes = {
    children: PropTypes.object,
  };

  return rtlRender(component, { wrapper: Wrapper });
}

export * from '@testing-library/react';
export default render;
