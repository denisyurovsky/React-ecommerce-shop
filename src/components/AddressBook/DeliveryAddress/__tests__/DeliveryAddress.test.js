import userEvent from '@testing-library/user-event';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import React from 'react';

import { addresses } from '../../../../test-utils/dto/addressesDto';
import { citiesRU, citiesBY } from '../../../../test-utils/dto/citiesDto';
import { countries } from '../../../../test-utils/dto/countriesDto';
import renderWith, { screen } from '../../../../test-utils/renderWith';
import DeliveryAddress from '../DeliveryAddress';

const preloaderState = {
  countries: {
    data: [],
    isLoading: false,
    errorOccurred: false,
    errorMessage: '',
  },
};

const server = setupServer(
  rest.get('/countries', (req, res, ctx) =>
    res(ctx.json(countries), ctx.status(200))
  ),
  rest.get('/cities/RU', (req, res, ctx) => res(ctx.json(citiesRU))),
  rest.get('/cities/BY', (req, res, ctx) => res(ctx.json(citiesBY)))
);

describe('DeliveryAddress component', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('should render valid snapshot', async () => {
    const { asFragment } = renderWith(
      <DeliveryAddress
        handleChange={() => {}}
        address={addresses[0]}
        setAddress={() => {}}
      />,
      preloaderState
    );

    const autocompleteCountry = screen.getByTestId('autocomplete-country');
    const countryInput = autocompleteCountry.querySelector('input');

    userEvent.type(countryInput, 'Bela');
    await screen.findByRole('option', { name: /Belarus/i });
    userEvent.type(autocompleteCountry, '{arrowdown}');
    userEvent.type(autocompleteCountry, '{enter}');

    expect(asFragment()).toMatchSnapshot();
  });
});
