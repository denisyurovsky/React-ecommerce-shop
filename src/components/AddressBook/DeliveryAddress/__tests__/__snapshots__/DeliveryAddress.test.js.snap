// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryAddress component should render valid snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper MuiBox-root css-0"
  >
    <div
      aria-expanded="false"
      class="MuiAutocomplete-root Mui-focused MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon country css-16awh2u-MuiAutocomplete-root"
      data-testid="autocomplete-country"
      role="combobox"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-focused css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="mui-2"
          id="mui-2-label"
        >
          Country
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-ghsjzk-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-autocomplete="list"
            aria-describedby="mui-2-helper-text"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="mui-2"
            spellcheck="false"
            type="text"
            value="Belarus"
          />
          <div
            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
              tabindex="-1"
              title="Clear"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium Mui-focused MuiFormHelperText-filled css-1d1r5q-MuiFormHelperText-root"
          id="mui-2-helper-text"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </p>
      </div>
    </div>
    <div
      aria-expanded="false"
      class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
      data-testid="autocomplete-city"
      role="combobox"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="mui-4"
          id="mui-4-label"
        >
          City
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-ghsjzk-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-autocomplete="list"
            aria-describedby="mui-4-helper-text"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="mui-4"
            spellcheck="false"
            type="text"
            value="Saint Petersburg"
          />
          <div
            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
              tabindex="-1"
              title="Clear"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-filled css-1d1r5q-MuiFormHelperText-root"
          id="mui-4-helper-text"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </p>
      </div>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="street"
      id="street-label"
    >
      Street
    </label>
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-describedby="street-helper-text"
        aria-invalid="false"
        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id="street"
        type="text"
        value="Pulkovskaya"
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-filled css-1d1r5q-MuiFormHelperText-root"
      id="street-helper-text"
    >
      <span
        class="notranslate"
      >
        ​
      </span>
    </p>
  </div>
  <div
    class="wrapper MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root MuiTextField-root building css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="building"
        id="building-label"
      >
        Building
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-describedby="building-helper-text"
          aria-invalid="false"
          class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          id="building"
          type="text"
          value="8"
        />
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-filled css-1d1r5q-MuiFormHelperText-root"
        id="building-helper-text"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </p>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="flat"
        id="flat-label"
      >
        Flat
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          id="flat"
          type="text"
          value="4"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="mui-5"
      id="mui-5-label"
    >
      Zip
    </label>
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-describedby="mui-5-helper-text"
        aria-invalid="false"
        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id="mui-5"
        type="text"
        value="168142"
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-filled css-1d1r5q-MuiFormHelperText-root"
      id="mui-5-helper-text"
    >
      <span
        class="notranslate"
      >
        ​
      </span>
    </p>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
